src/app/models/function.rb

```diff
@@ -14,6 +16,17 @@ class Function < ApplicationRecord
 
   has_and_belongs_to_many :tags
 
+  def tags_list
+    tags.map(&:name).join(', ')
+  end
+
+  def tags_list=(new_value)
+    # TODO: Raise error if number of tags excedded
+    #       the limit defined in TAGS_LIMIT
+    tag_names = new_value.split(',').reject(&:blank?)
+    self.tags = tag_names.map { |name| Tag.find_or_create_by(name: name.strip&.downcase) }
```

Comment:
can we change this to initialization, so we can validate the function before creating any tags for it

```suggestion
    self.tags = tag_names.map { |name| Tag.find_or_initialize_by(name: name.strip&.downcase) }
```

Reply:
for validation this solution seems like a good one https://stackoverflow.com/a/24496723/458436 it also needs a custom message for validation in locales